src/edu/stanford/nlp/mt/util/ParallelSuffixArray.java

```diff
@@ -83,10 +83,16 @@ public ParallelSuffixArray(String sourceFile, String targetFile, String alignFil
   public void write(Kryo kryo, Output output) {
     writeArray(srcBitext, output);
     writeArray(tgtBitext, output);
-    int[] e2fPrimitive = (int[]) org.apache.commons.lang3.ArrayUtils.toPrimitive(e2f);
-    writeArray(e2fPrimitive, output);
-    int[] f2ePrimitive = (int[]) org.apache.commons.lang3.ArrayUtils.toPrimitive(f2e);
-    writeArray(f2ePrimitive, output);
+    writeArray(new int[]{e2f.length}, output);
```

Comment:
I think it would it be simpler to do
```
output.writeInt(e2f.length, true);
```

Optionally, pull this line and the loop below it out into a `writeArrayOfSets` method.

Reply:
This would be much cleaner.